pkg/landscaper/controllers/execution/controller.go

```diff
@@ -169,3 +179,28 @@ func HandleErrorFunc(log logr.Logger, client client.Client, eventRecorder record
 		return err
 	}
 }
+
+// handleDeployItemPhaseChanges updates the phase of the given execution, if its phase doesn't match the combined phase of its deploy items anymore
+// it returns a boolean indicating whether the phase was changed
+func handleDeployItemPhaseChanges(ctx context.Context, op *execution.Operation, logger logr.Logger, exec *lsv1alpha1.Execution) (bool, error) {
+	deployitems, err := op.ListManagedDeployItems(ctx)
+	if err != nil {
+		return false, lserrors.NewWrappedError(err, "Reconcile", "ListManagedDeployItems", err.Error())
+	}
+	phases := []lsv1alpha1.ExecutionPhase{}
+	for _, di := range deployitems {
+		phases = append(phases, di.Status.Phase)
+	}
+	cp := lsv1alpha1helper.CombinedExecutionPhase(phases...)
+	if exec.Status.Phase != cp {
+		// Phase is completed but doesn't fit to the deploy items' phases
+		logger.V(5).Info("execution phase mismatch", "phase", string(exec.Status.Phase), "combinedPhase", string(cp))
+		exec.Status.Phase = cp
```

Comment:
What if a deploy item changes from failed to succeeded and exports new data.
So I guess we also have to update the execution export?

Reply:
Damn, you are right ...